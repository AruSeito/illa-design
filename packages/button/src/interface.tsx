import { ButtonHTMLAttributes, HTMLAttributes, ReactNode } from "react"

export type ButtonColorScheme =
  "white"
  | "blackAlpha"
  | "gray"
  | "grayBlue"
  | "red"
  | "orange"
  | "yellow"
  | "green"
  | "blue"
  | "cyan"
  | "purple"
  | "techBlue"

export type ButtonSize = "small" | "medium" | "large"
export type ButtonVariant = "fill" | "dashed" | "outline" | "text"
export type ButtonShape = "square" | "round"

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  colorScheme?: ButtonColorScheme
  size?: ButtonSize
  variant?: ButtonVariant
  shape?: ButtonShape
  fullWidth?: boolean
  loading?: boolean
  loadingText?: string
  leftIcon?: ReactNode
  rightIcon?: ReactNode
}
