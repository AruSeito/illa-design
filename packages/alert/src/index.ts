export * from "./alert"
export * from "./interface"
