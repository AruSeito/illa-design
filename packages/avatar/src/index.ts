export * from "./avatar"
export * from "./interface"
export * from "./avatar-group"