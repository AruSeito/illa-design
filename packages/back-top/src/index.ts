export * from "./backtop"
export * from "./interface"
export * from "./style"
