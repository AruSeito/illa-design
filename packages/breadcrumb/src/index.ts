export * from "./interface"
export * from "./breadcrumb"
export * from "./breadcrumbItem"
