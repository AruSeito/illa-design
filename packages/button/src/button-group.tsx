/** @jsxImportSource @emotion/react */
import * as React from "react"
import { Children, createContext, forwardRef, ReactNode } from "react"
import { ButtonGroupContextProps, ButtonGroupProps } from "./interface"
import { applySpacing, avatarGroupContainer } from "./style"

export const ButtonGroupContext = createContext<ButtonGroupContextProps | undefined>(undefined)

export const ButtonGroup = forwardRef<HTMLDivElement, ButtonGroupProps>((props, ref) => {
  const {
    colorScheme = "blue",
    size = "small",
    variant = "fill",
    shape = "square",
    spacing = "4px",
    attached = false,
    ...otherProps
  } = props
  const newChildren = Children.map(props.children, (child: ReactNode, index: number) => {
    const first = index == 0
    const last = index == Children.count(props.children) - 1
    return <ButtonGroupContext.Provider value={{ colorScheme, size, variant, shape, spacing, attached, first, last }}>
      <span css={applySpacing(spacing, index, attached)}>
        {child}
      </span>
    </ButtonGroupContext.Provider>
  })
  return <div css={avatarGroupContainer} ref={ref} {...otherProps}>
    {newChildren}
  </div>
})