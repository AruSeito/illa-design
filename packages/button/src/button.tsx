/** @jsxImportSource @emotion/react */
import { Children, forwardRef } from "react"
import { ButtonColorScheme, ButtonProps, ButtonShape, ButtonSize, ButtonVariant } from "./interface"
import { css, SerializedStyles } from "@emotion/react"
import { globalColor, illaPrefix } from "@illa-design/theme"
import { LoadingIcon } from "@illa-design/icon"

const tagContainerCss = css`
  vertical-align: middle;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
`

enum State {
  DEFAULT,
  HOVER,
  FOCUSED,
  DISABLE
}

function getDifferentStatusColor(colorScheme: ButtonColorScheme, variant: ButtonVariant, state: State): string[] {
  switch (state) {
    case State.DEFAULT:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-01`), globalColor(`--${illaPrefix}-white-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-09`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "dashed":
        case "outline":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-01`), globalColor(`--${illaPrefix}-${colorScheme}-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-08`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "text":
          return [globalColor("transparent"), globalColor(`--${illaPrefix}-${colorScheme}-01`)]
      }
    case State.HOVER:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-02`), globalColor(`--${illaPrefix}-white-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-08`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "dashed":
        case "outline":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-02`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-07`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "text":
          return [globalColor("transparent"), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
      }
    case State.FOCUSED:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-n-01`), globalColor(`--${illaPrefix}-white-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-06`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "dashed":
        case "outline":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-n-01`), globalColor(`--${illaPrefix}-${colorScheme}-n-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-05`), globalColor(`--${illaPrefix}-${colorScheme}-02`)]
          }
        case "text":
          return [globalColor("transparent"), globalColor(`--${illaPrefix}-${colorScheme}-n-01`)]
      }
    case State.DISABLE:
      switch (variant) {
        case "fill":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-05`), globalColor(`--${illaPrefix}-white-01`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-09`), globalColor(`--${illaPrefix}-${colorScheme}-05`)]
          }
        case "dashed":
        case "outline":
          if (colorScheme != "gray") {
            return [globalColor(`--${illaPrefix}-${colorScheme}-05`), globalColor(`--${illaPrefix}-${colorScheme}-05`)]
          } else {
            return [globalColor(`--${illaPrefix}-${colorScheme}-08`), globalColor(`--${illaPrefix}-${colorScheme}-05`)]
          }
        case "text":
          return [globalColor("transparent"), globalColor(`--${illaPrefix}-${colorScheme}-05`)]
      }
  }
}

function applyBg(variant: ButtonVariant, colorScheme: ButtonColorScheme): SerializedStyles {
  switch (variant) {
    case "text":
      return css``
    case "dashed":
      return css`
        border: dashed 1px ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[0]};

        &:hover {
          border: dashed 1px ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[0]};
        }

        &:active {
          border: dashed 1px ${getDifferentStatusColor(colorScheme, variant, State.FOCUSED)[0]};
        }

        &:disabled {
          border: dashed 1px ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[0]};
        }
      `
    case "fill":
      return css`
        background-color: ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[0]};

        &:hover {
          background-color: ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[0]};
        }

        &:active {
          background-color: ${getDifferentStatusColor(colorScheme, variant, State.FOCUSED)[0]};
        }

        &:disabled {
          background-color: ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[0]};
        }
      `
    case "outline":
      return css`
        border: solid 1px ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[0]};

        &:hover {
          border: solid 1px ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[0]};
        }

        &:active {
          border: solid 1px ${getDifferentStatusColor(colorScheme, variant, State.FOCUSED)[0]};
        }

        &:disabled {
          border: solid 1px ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[0]};
        }
      `
  }
}

function applyCursor(loading: boolean, disabled: boolean): SerializedStyles {
  if (loading) {
    return css`
      cursor: default;
    `
  } else if (disabled) {
    return css`
      cursor: not-allowed;
    `
  } else {
    return css`
      cursor: pointer;
    `
  }
}

function applyElementColor(variant: ButtonVariant, colorScheme: ButtonColorScheme): SerializedStyles {
  switch (variant) {
    case "text":
    case "outline":
    case "dashed":
      return css`
        color: ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[1]};

        &:hover {
          color: ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[1]};
        }

        &:active {
          color: ${getDifferentStatusColor(colorScheme, variant, State.FOCUSED)[1]};
        }

        &:disabled {
          color: ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[1]};
        }
      `
    case "fill":
      return css`
        color: ${getDifferentStatusColor(colorScheme, variant, State.DEFAULT)[1]};

        &:hover {
          color: ${getDifferentStatusColor(colorScheme, variant, State.HOVER)[1]};
        }

        &:active {
          color: ${getDifferentStatusColor(colorScheme, variant, State.FOCUSED)[1]};
        }

        &:disabled {
          color: ${getDifferentStatusColor(colorScheme, variant, State.DISABLE)[1]};
        }
      `
  }
}

function applyShape(shape: ButtonShape): SerializedStyles {
  switch (shape) {
    case "square":
      return css`
        border-radius: 4px;
      `
    case "round":
      return css`
        border-radius: 999px;
      `
  }
}

function applyPaddingStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        padding: 2px 12px;
      `
    case "medium":
      return css`
        padding: 5px 16px;
      `
    case "large":
      return css`
        padding: 9px 16px;
      `
  }
}

function applyWithoutTextSize(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        justify-content: center;
        width: 24px;
        height: 24px;
      `
    case "medium":
      return css`
        justify-content: center;
        width: 32px;
        height: 32px;
      `
    case "large":
      return css`
        justify-content: center;
        width: 40px;
        height: 40px;
      `
  }
}

function applyFontStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        font-size: 12px;
        font-weight: normal;
        font-stretch: normal;
        font-style: normal;
        line-height: 20px;
        letter-spacing: normal;
      `
    case "medium":
    case "large":
      return css`
        font-size: 14px;
        font-weight: normal;
        font-stretch: normal;
        font-style: normal;
        line-height: 22px;
        letter-spacing: normal;
      `
  }
}

function applyLeftIconStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        margin-right: 6px;
      `
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        margin-right: 8px;
      `
  }
}

function applyIconWithoutText(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        justify-content: center;
      `
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        justify-content: center;
      `
  }
}

function applyRightIconStyle(size: ButtonSize): SerializedStyles {
  switch (size) {
    case "small":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        margin-left: 6px;
      `
    case "medium":
    case "large":
      return css`
        display: inline-flex;
        flex-direction: row;
        align-items: center;
        width: 12px;
        height: 12px;
        margin-left: 8px;
      `
  }
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {

  const {
    colorScheme = "blue",
    size = "small",
    variant = "fill",
    shape = "square",
    fullWidth = false,
    loading = false,
    loadingText = undefined,
    leftIcon = undefined,
    disabled = false,
    rightIcon = undefined,
    ...otherProps
  } = props

  const hasLoadingText = loadingText != undefined && loadingText.length > 0
  const hasChild = Children.count(props.children) >= 1

  if (hasChild || (hasLoadingText && loading)) {

    const finalContainer = css`
      ${tagContainerCss};
      ${applyCursor(loading, disabled)}
      ${applyPaddingStyle(size)};
      ${applyShape(shape)};
      ${applyElementColor(variant, colorScheme)};
      ${applyBg(variant, colorScheme)};
    `

    return <button ref={ref} css={finalContainer} {...otherProps} disabled={disabled || loading}>
      {(loading || leftIcon) && <span css={applyLeftIconStyle(size)}>
      {loading ? <LoadingIcon spin={true} /> : leftIcon}
    </span>}
      <span css={applyFontStyle(size)}>
      {loading && loadingText ? loadingText : props.children}
    </span>
      {rightIcon && <span css={applyRightIconStyle(size)}>
      {rightIcon}
    </span>}
    </button>
  } else {

    const finalContainer = css`
      ${tagContainerCss};
      ${applyCursor(loading, disabled)}
      ${applyWithoutTextSize(size)};
      ${applyShape(shape)};
      ${applyElementColor(variant, colorScheme)};
      ${applyBg(variant, colorScheme)};
    `

    return <button ref={ref} css={finalContainer} {...otherProps} disabled={disabled || loading}>
      {
        (loading || leftIcon) &&
        <span css={applyIconWithoutText(size)}>
          {
            loading ? <LoadingIcon spin={true} /> : leftIcon
          }
        </span>
      }
      {
        rightIcon &&
        <span css={applyIconWithoutText(size)}>
          {rightIcon}
        </span>
      }
    </button>
  }

})
