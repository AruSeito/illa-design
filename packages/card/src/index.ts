export * from "./card"
export * from "./interface"
export * from "./grid"
export * from "./meta"
