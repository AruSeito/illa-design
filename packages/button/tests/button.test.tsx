import { Button } from "../src"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { BsApp, BsTwitch } from "react-icons/bs"

test("Button renders with text", () => {
  render(<Button data-testid="test">Hello</Button>)
  expect(screen.getByTestId("test")).toBeInTheDocument()
})

test("Button renders with different colorscheme", () => {
  render(<Button data-testid="test-with-gray" colorScheme="gray">Hello</Button>)
  expect(screen.getByTestId("test-with-gray")).toBeInTheDocument()
  render(<Button data-testid="test-with-green" colorScheme="green">Hello</Button>)
  expect(screen.getByTestId("test-with-green")).toBeInTheDocument()
})

test("Button renders without text", () => {
  render(<Button data-testid="test-without-text" />)
  expect(screen.getByTestId("test-without-text")).toBeInTheDocument()
  render(<Button data-testid="test-without-text-medium" size="medium" rightIcon={<BsApp />} />)
  expect(screen.getByTestId("test-without-text-medium")).toBeInTheDocument()
  render(<Button data-testid="test-without-text-large" size="large" rightIcon={<BsApp />} />)
  expect(screen.getByTestId("test-without-text-large")).toBeInTheDocument()
})

test("Button renders with different size", () => {
  render(<Button data-testid="test-small" size="small" rightIcon={<BsTwitch />} leftIcon={<BsApp />}>Hello</Button>)
  expect(screen.getByTestId("test-small")).toBeInTheDocument()
  render(<Button data-testid="test-medium" size="medium" rightIcon={<BsTwitch />} leftIcon={<BsApp />}>Hello</Button>)
  expect(screen.getByTestId("test-medium")).toBeInTheDocument()
  render(<Button data-testid="test-large" size="large" rightIcon={<BsTwitch />} leftIcon={<BsApp />}>Hello</Button>)
  expect(screen.getByTestId("test-large")).toBeInTheDocument()
})

test("Button renders with different variant", () => {
  render(<Button data-testid="test-with-fill" variant="fill">Hello</Button>)
  expect(screen.getByTestId("test-with-fill")).toMatchSnapshot()
  render(<Button data-testid="test-with-dashed" variant="dashed">Hello</Button>)
  expect(screen.getByTestId("test-with-dashed")).toMatchSnapshot()
  render(<Button data-testid="test-with-dashed-gray" variant="dashed" colorScheme="gray">Hello</Button>)
  expect(screen.getByTestId("test-with-dashed-gray")).toMatchSnapshot()
  render(<Button data-testid="test-with-outline" variant="outline">Hello</Button>)
  expect(screen.getByTestId("test-with-outline")).toMatchSnapshot()
  render(<Button data-testid="test-with-outline-gray" variant="outline" colorScheme="gray">Hello</Button>)
  expect(screen.getByTestId("test-with-outline-gray")).toMatchSnapshot()
  render(<Button data-testid="test-with-text" variant="text">Hello</Button>)
  expect(screen.getByTestId("test-with-text")).toMatchSnapshot()
})

test("Button renders round", () => {
  render(<Button data-testid="test-with-round" shape="round">Hello</Button>)
  expect(screen.getByTestId("test-with-round")).toMatchSnapshot()
})

test("Button renders fullwidth", () => {
  render(<div style={{
    width: "200px",
  }}>
    <Button data-testid="test-with-fullwidth" fullWidth={true}>Hello</Button>
  </div>)
  expect(screen.getByTestId("test-with-fullwidth")).toMatchSnapshot()
})