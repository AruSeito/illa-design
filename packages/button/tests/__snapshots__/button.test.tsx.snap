// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders fullwidth 1`] = `
.emotion-0 {
  width: 100%;
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 2px 12px;
  border-radius: 4px;
  color: #fff;
  background-color: #2a3ed6;
}

.emotion-0:hover {
  color: #fff;
}

.emotion-0:active {
  color: #fff;
}

.emotion-0:disabled {
  color: #fff;
}

.emotion-0:hover {
  background-color: #3e53eb;
}

.emotion-0:active {
  background-color: #1e29bd;
}

.emotion-0:disabled {
  background-color: #aebeff;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-fullwidth"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders round 1`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 2px 12px;
  border-radius: 999px;
  color: #fff;
  background-color: #2a3ed6;
}

.emotion-0:hover {
  color: #fff;
}

.emotion-0:active {
  color: #fff;
}

.emotion-0:disabled {
  color: #fff;
}

.emotion-0:hover {
  background-color: #3e53eb;
}

.emotion-0:active {
  background-color: #1e29bd;
}

.emotion-0:disabled {
  background-color: #aebeff;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-round"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 1`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 2px 12px;
  border-radius: 4px;
  color: #fff;
  background-color: #2a3ed6;
}

.emotion-0:hover {
  color: #fff;
}

.emotion-0:active {
  color: #fff;
}

.emotion-0:disabled {
  color: #fff;
}

.emotion-0:hover {
  background-color: #3e53eb;
}

.emotion-0:active {
  background-color: #1e29bd;
}

.emotion-0:disabled {
  background-color: #aebeff;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-fill"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 2`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1px 11px;
  border-radius: 4px;
  color: #2a3ed6;
  border: dashed 1px #2a3ed6;
}

.emotion-0:hover {
  color: #3e53eb;
}

.emotion-0:active {
  color: #1e29bd;
}

.emotion-0:disabled {
  color: #aebeff;
}

.emotion-0:hover {
  border: dashed 1px #3e53eb;
}

.emotion-0:active {
  border: dashed 1px #1e29bd;
}

.emotion-0:disabled {
  border: dashed 1px #aebeff;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-dashed"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 3`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1px 11px;
  border-radius: 4px;
  color: #1f1f1f;
  border: dashed 1px #ebebeb;
}

.emotion-0:hover {
  color: #1f1f1f;
}

.emotion-0:active {
  color: #1f1f1f;
}

.emotion-0:disabled {
  color: #c2c2c2;
}

.emotion-0:hover {
  border: dashed 1px #e0e0e0;
}

.emotion-0:active {
  border: dashed 1px #c2c2c2;
}

.emotion-0:disabled {
  border: dashed 1px #ebebeb;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-dashed-gray"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 4`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1px 11px;
  border-radius: 4px;
  color: #2a3ed6;
  border: solid 1px #2a3ed6;
}

.emotion-0:hover {
  color: #3e53eb;
}

.emotion-0:active {
  color: #1e29bd;
}

.emotion-0:disabled {
  color: #aebeff;
}

.emotion-0:hover {
  border: solid 1px #3e53eb;
}

.emotion-0:active {
  border: solid 1px #1e29bd;
}

.emotion-0:disabled {
  border: solid 1px #aebeff;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-outline"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 5`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 1px 11px;
  border-radius: 4px;
  color: #1f1f1f;
  border: solid 1px #ebebeb;
}

.emotion-0:hover {
  color: #1f1f1f;
}

.emotion-0:active {
  color: #1f1f1f;
}

.emotion-0:disabled {
  color: #c2c2c2;
}

.emotion-0:hover {
  border: solid 1px #e0e0e0;
}

.emotion-0:active {
  border: solid 1px #c2c2c2;
}

.emotion-0:disabled {
  border: solid 1px #ebebeb;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-outline-gray"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;

exports[`Button renders with different variant 6`] = `
.emotion-0 {
  vertical-align: middle;
  white-space: nowrap;
  outline: none;
  margin: 0;
  border: 0;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 2px 12px;
  border-radius: 4px;
  color: #2a3ed6;
}

.emotion-0:disabled {
  color: #aebeff;
}

.emotion-0:hover {
  background-color: #f5f5f5;
}

.emotion-0:active {
  background-color: #ebebeb;
}

.emotion-1 {
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 20px;
  letter-spacing: normal;
}

<button
  class="emotion-0"
  data-testid="test-with-text"
>
  <span
    class="emotion-1"
  >
    Hello
  </span>
</button>
`;
